src/warnings.js

```diff
@@ -54,11 +54,11 @@ export const warnings = Object.freeze({
 });
 
 export function generateWarning(kind, options) {
-  const { location, file = null, value = null } = options;
+  const { location, file = null, value = null, source = "JS-X-Ray" } = options;
 
   if (kind === "encoded-literal") {
     return Object.assign(
-      { kind, value, location: [utils.toArrayLocation(location)] },
+      { kind, value, location: [utils.toArrayLocation(location)], source },
```

Comment:
You need to add source to the `result` Object too (few lines below).

Reply:
@fabnguess you missed the `result` const line 66